renderdoc/driver/d3d11/d3d11_shaderdebug.cpp

```diff
@@ -96,6 +99,302 @@ void D3D11DebugAPIWrapper::AddDebugMessage(MessageCategory c, MessageSeverity sv
   m_pDevice->AddDebugMessage(c, sv, src, d);
 }
 
+bool D3D11DebugAPIWrapper::FetchSRV(const ShaderDebug::BindingSlot &slot)
+{
+  RDCASSERT(slot.registerSpace == 0);
+  RDCASSERT(slot.shaderRegister < D3D11_COMMONSHADER_INPUT_RESOURCE_SLOT_COUNT);
+
+  D3D11RenderState *rs = m_pDevice->GetImmediateContext()->GetCurrentPipelineState();
+  ID3D11ShaderResourceView *pSRV = NULL;
+  if(GetShaderType() == DXBC::ShaderType::Vertex)
+    pSRV = rs->VS.SRVs[slot.shaderRegister];
+  else if(GetShaderType() == DXBC::ShaderType::Pixel)
+    pSRV = rs->PS.SRVs[slot.shaderRegister];
+  else if(GetShaderType() == DXBC::ShaderType::Compute)
+    pSRV = rs->CS.SRVs[slot.shaderRegister];
+
+  if(!pSRV)
+    return false;
+
+  ID3D11Resource *res = NULL;
+  pSRV->GetResource(&res);
+
+  D3D11_SHADER_RESOURCE_VIEW_DESC sdesc;
+  pSRV->GetDesc(&sdesc);
+
+  ShaderDebug::GlobalState::SRVData srvData;
```

Comment:
This might be better to write as a reference directly into the map, to force its allocation first and avoid copying from the temporary?

Reply:
Good call, fixed.